Feature/UI login page test

AnastasiyaSoroka opened the pull request:
The following updated
 - pages (BasePage, LoginPage, ProjectsListPage)
 - tests (BaseTest, LoginTest)
 - config (added token, urls)
 - README was renamed
 - circle.ci

Commit:
Adding circleci

Commit:
Adding circleci

Commit:
Some updates for circle ci

Commit:
Merge remote-tracking branch 'origin/feature/ui-login-page-test' into feature/ui-login-page-test

# Conflicts:
#	src/test/java/tests/BaseTest.java
#	src/test/java/utils/CapabilitiesGenerator.java

Commit:
Some updates for circle ci

Commit:
Some updates for circle ci

Commit:
Some updates for circle ci

Commit:
Some updates for circle ci

Commit:
Some updates for circle ci

Commit:
Some updates for circle ci

Commit:
Added config.properties

Commit:
Some updates

Commit:
The following updated
 - pages (BasePage, LoginPage, ProjectsListPage)
 - tests (BaseTest, LoginTest)
 - config (added token, urls)
 - README was renamed
 - circle.ci

Commit:
Updated LoginTest

dzmitryrak commented on src/test/java/pages/LoginPage.java:
```diff
@@ -0,0 +1,64 @@
+        return this;
+    }
+
+    @Step("Trying to log in with the following parameters:")
```
Так. Стоп. Тут уже не отвертишься. Давай-ка через 3-х слойную структуру. TESTS - STEPS - PAGES

AnastasiyaSoroka replied:
не прокатило(

dzmitryrak commented on src/test/java/pages/LoginPage.java:
```diff
@@ -0,0 +1,64 @@
+        driver.findElement(EMAIL_INPUT).sendKeys(username);
+        log.info("Populating Password with: " + password);
+        driver.findElement(PASSWORD_INPUT).sendKeys(password);
+        log.info("Clicking on Login button");
```
Слишком подробные логи уже пошли

dzmitryrak commented on src/test/java/pages/LoginPage.java:
```diff
@@ -0,0 +1,64 @@
+    @Step("Error appears")
+    public String getErrorText() {
+        isErrorAppears();
+        AllureUtils.takeScreenshot(driver);
```
Таких скриншотов не надо. Только в TestLister

dzmitryrak commented on src/test/java/tests/BaseTest.java:
```diff
@@ -0,0 +1,50 @@
+        loginPage = new LoginPage(driver);
+        projectsListPage = new ProjectsListPage(driver);
+
+        log.info("Setting driver into context with variable name " + variable);
```
этот лог не нужен

dzmitryrak commented on src/test/java/tests/BaseTest.java:
```diff
@@ -0,0 +1,50 @@
+        context.setAttribute(variable, driver);
+    }
+
+    @AfterMethod
```
alwaysRUn=true

dzmitryrak commented on src/test/java/tests/BaseTest.java:
```diff
@@ -0,0 +1,50 @@
+
+    @AfterMethod
+    public void tearDown() {
+        driver.quit();
```
проверку на null тут сделай

dzmitryrak commented on src/test/java/utils/AllureUtils.java:
```diff
@@ -0,0 +1,13 @@
+import org.openqa.selenium.TakesScreenshot;
+import org.openqa.selenium.WebDriver;
+
+public class AllureUtils {
```
В принципе не нужен класс. Ну или разве что переиспользовать его внутри TestListener

dzmitryrak commented on src/test/java/utils/TestListener.java:
```diff
@@ -0,0 +1,57 @@
+    private byte[] takeScreenshot(ITestResult iTestResult) {
+        ITestContext context = iTestResult.getTestContext();
+        try {
+            return ((TakesScreenshot) context.getAttribute("driver")).getScreenshotAs(OutputType.BYTES);
```
проверку засунь еще если драйвер null

Commit:
Comments were fixed
 - steps for LoginPage
 - removed extra logs, screenshots
 - removed AllureUtils.java
 - some null validation was added
